### ISSUE TITLE
Provide complete CRD example in the docs

### ISSUE THREAD

**Issue by mumrau** [jaegertracing/jaeger-operator#1408]
Hello,

I do think that for every CRD proposed by every operator in the kubernetes world, we need a _complete and exhaustive_ "example" of documetantion regarding how we can populate the CRD. I haven't found such documentation neither in the `jaeger-operator`, the helm chart counterpart, neither on the official documentation found at https://www.jaegertracing.io/docs/1.22/operator/ which contains very sparse information about how to populate the fields.

It would also be incredibly helpful in order to help the `jaegers` instance definitions to be kept up to date with the upgrades.

Maybe I have missed out on where to find such documentation, but having a complete yaml file with _every single field consumed by the operator_ explicitly stated would probably be a great addition.

**Comment by jpkrohling** [comment]
You are absolutely right that we need better documentation on our CRD. Are you aware of any tools that are able to do that based on the comments we have in the source code? Just doing it once will certainly be a maintenance headache, as things will get out of sync too frequently, I'm afraid.

**Comment by mumrau** [comment]
Sadly I am not aware of such tooling and oh god.... it can be extremely complicated. Compatibility matrix should be kept up to date as well, documentation as well.
I just wanted to raise that up since I am having trouble keeping my CRD up to date (and not solely for jaeger, other projects lack similar resources).

Nonetheless, I don't think the CRD changed _that much_ recently, maybe maintaining it manually makes sense as well, even though I would rather rely on automation for sure.

**Comment by jpkrohling** [comment]
I have in my queue somewhere that I need to check how Kubernetes itself generates this doc. I think they have custom tooling that we can reuse.

**Comment by ethernoy** [comment]
Documentation on each field in the Jaeger CRD is definitely wanted

**Comment by pavolloffay** [comment]
This is duplicate for #1248
